dynamodb/src/main/scala/awscala/dynamodbv2/Table.scala

```diff
@@ -70,80 +69,68 @@ case class Table(
-        .flatMap { case (_, getterValue) => Some(getterValue) }
-
-      found.map { f =>
-        AnnotationMeta(
-          a.name,
-          a.annotation,
-          {
-            a.typeSignature match {
-              case "Int" => f.asInstanceOf[Int]
-              case "String" => f.asInstanceOf[String]
-            }
-          })
+    val constructorArgs: Seq[AnnotatedConstructorArgMeta] = extractAnnotatedConstructorArgs(entity)
+    val getterCallResults: Seq[(String, AnyRef)] = extractGetterNameAndValue(entity)
+
+    var maybeHashPK: Option[Any] = None
+    var maybeRangePK: Option[Any] = None
+    val attributes: ListBuffer[(String, AnyRef)] = ListBuffer()
+    for (arg <- constructorArgs) {
+      getterCallResults.find { case (name, _) => name == arg.name } match {
+        case Some((_, value)) if arg.annotationNames.contains("hashPK") => maybeHashPK = Some(value)
+        case Some((_, value)) if arg.annotationNames.contains("rangePK") => maybeRangePK = Some(value)
+        case Some(nameAndValue) => attributes += nameAndValue
+        case _ => // noop
       }
-    })
-
-    val hashKey: Option[Any] = annotationFields
-      .find(a => a.annotation.contains("hashPK"))
-      .map(a => a.typeSignature)
-
-    val rangeKey: Option[Any] = annotationFields
-      .find(a => a.annotation.contains("rangePK"))
-      .map(a => a.typeSignature)
-
-    val attributes = fields.filter {
-      case (getterName, _) =>
-        !annotationFields.exists(a => getterName == a.name)
     }
-
-    if (hashKey.isEmpty)
-      throw new Exception(s"Primary key is not defined for ${entity.getClass.getName}")
-
-    if (hashKey.isDefined && rangeKey.isDefined)
-      dynamoDB.put(this, hashKey.get, rangeKey.get, attributes: _*)
-    else if (hashKey.isDefined)
-      dynamoDB.put(this, hashKey.get, attributes: _*)
+    (maybeHashPK, maybeRangePK) match {
+      case (Some(hashPK), Some(rangePK)) =>
+        dynamoDB.put(this, hashPK, rangePK, attributes.toSeq: _*)
+      case (Some(hashPK), None) =>
+        dynamoDB.put(this, hashPK, attributes.toSeq: _*)
+      case _ =>
+        throw new Exception(s"Primary key is not defined for ${entity.getClass.getName}")
+    }
   }
 
-  def putItem[E <: AnyRef](hashPK: Any, rangePK: Any, entity: E)(implicit dynamoDB: DynamoDB): Unit = {
-    val attrs = getterNamesFromEntity(entity)
-    dynamoDB.put(this, hashPK, rangePK, attrs: _*)
-  }
+  private case class AnnotatedConstructorArgMeta(name: String, annotationNames: Seq[String])
 
-  private def getterAnnotationsFromEntity[T: u.TypeTag](entity: T): List[AnnotationMeta] = {
+  private def extractAnnotatedConstructorArgs[T: u.TypeTag](entity: T): Seq[AnnotatedConstructorArgMeta] = {
     u.typeOf[entity.type].decl(termNames.CONSTRUCTOR).asMethod.paramLists.flatten
       .collect({
-        case t: TermSymbol if (t.isVal && t.annotations.nonEmpty) =>
-          AnnotationMeta(t.name.toString, t.annotations.head.toString, t.typeSignature.toString)
```

Comment:
`t.annotations.head.toString` does not work if the field have a different annotation as the first one.